base_kanban_stage/models/base_kanban_abstract.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from openerp import api, fields, models
+
+
+class BaseKanbanAbstract(models.AbstractModel):
+    '''Inherit from this class to add support for Kanban stages to your model.
+    All public properties are preceded with kanban_ in order to isolate from
+    child models, with the exception of stage_id, which is a required field in
+    the Kanban widget and must be defined as such.'''
+
+    _name = 'base.kanban.abstract'
+    _order = 'kanban_priority desc, kanban_sequence'
+    _group_by_full = {
+        'stage_id': lambda s, *a, **k: s._read_group_stage_ids(*a, **k),
+    }
+
+    kanban_sequence = fields.Integer(
+        default=10,
+        index=True,
+        help='Order of record in relation to other records in the same Kanban'
+             ' stage and with the same priority',
+    )
+    kanban_priority = fields.Selection(
+        selection=[('0', 'Normal'), ('5', 'Medium'), ('10', 'High')],
+        index=True,
+        default='0',
+        help='The priority of the record (shown as stars in Kanban views)',
+    )
+    stage_id = fields.Many2one(
+        string='Kanban Stage',
+        comodel_name='base.kanban.stage',
+        track_visibility='onchange',
+        index=True,
+        copy=False,
+        help='The Kanban stage that this record is currently in',
+        default=lambda s: s._default_stage_id(),
+        domain=lambda s: [('res_model.model', '=', s._name)],
+    )
+    kanban_user_id = fields.Many2one(
```

Comment:
Can we use `user_id` directly? AFAIK it is the field name expected for the responsible person (in places like the `mail` module logic).